src/OneDriveSdk/Requests/UploadChunkRequest.cs

```diff
@@ -0,0 +1,136 @@
+        /// </summary>
+        /// <param name="stream">Stream of data to be sent in the request. Length must be equal to the length
+        /// of this chunk (as defined by this.RangeLength)</param>
+        /// <param name="cancellationToken">The cancellation token</param>
+        /// <returns>The status of the upload. If UploadSession.AdditionalData.ContainsKey("successResponse")
+        /// is true, then the item has completed, and the value is the created item from the server.</returns>
+        public async Task<UploadChunkResult> PutAsync(Stream stream, CancellationToken cancellationToken)
+        {
+            this.Method = "PUT";
+            using (var response = await this.SendRequestAsync(stream, cancellationToken).ConfigureAwait(false))
+            {
+                if (response.Content != null)
+                {
+                    var responseString = await response.Content.ReadAsStringAsync();
+                    
+                    if (response.StatusCode == HttpStatusCode.Created || response.StatusCode == HttpStatusCode.OK)
+                    {
+                        return new UploadChunkResult
+                            {
+                                ItemResponse =
+                                    this.Client.HttpProvider.Serializer.DeserializeObject<Item>(responseString)
+                            };
+                    }
+                    else
+                    {
+                        return new UploadChunkResult
+                            {
+                                UploadSession =
+                                    this.Client.HttpProvider.Serializer.DeserializeObject<UploadSession>(responseString)
+                            };
+                    }
+                }
+
+                return default(UploadChunkResult);
+            }
+        }
+
+        private async Task<HttpResponseMessage> SendRequestAsync(
+            Stream stream,
+            CancellationToken cancellationToken,
+            HttpCompletionOption completionOption = HttpCompletionOption.ResponseContentRead)
+        {
+            if (string.IsNullOrEmpty(this.RequestUrl))
+            {
+                throw new ArgumentNullException(nameof(this.RequestUrl), "Session Upload URL cannot be null or empty.");
+            }
+
+            if (this.Client.AuthenticationProvider == null)
+            {
+                throw new ArgumentNullException(nameof(this.Client.AuthenticationProvider), "Client.AuthenticationProvider must not be null.");
+            }
+
+            using (var request = this.GetHttpRequestMessage())
+            {
+                await this.Client.AuthenticationProvider.AuthenticateRequestAsync(request).ConfigureAwait(false);
+
+                request.Content = new StreamContent(stream);
+                request.Content.Headers.ContentRange =
+                    ContentRangeHeaderValue.Parse($"bytes {this.RangeBegin}-{this.RangeEnd}/{this.TotalSessionLength}");
```

Comment:
>ContentRangeHeaderValue [](start = 20, length = 23)

Use the constructor that takes three longs instead of having to parse a string, [https://msdn.microsoft.com/en-us/library/hh138063(v=vs.118).aspx](https://msdn.microsoft.com/en-us/library/hh138063(v=vs.118).aspx)